src/chttpd.erl

```diff
@@ -535,7 +535,15 @@ json_body_obj(Httpd) ->
 
 
 doc_etag(#doc{revs={Start, [DiskRev|_]}}) ->
-    "\"" ++ ?b2l(couch_doc:rev_to_str({Start, DiskRev})) ++ "\"".
+    rev_etag({Start, DiskRev}).
+
+rev_etag({Start, DiskRev}) ->
+    Rev = couch_doc:rev_to_str({Start, DiskRev}),
+    <<"\"", Rev/binary, "\"">>.
+
+make_etag() ->
```

Comment:
@rnewson: I don't think changing how `views`  handle etag generation is in the scope of the bug. I could work on it in the context of another bug if you think it needs to be done before 2.0 release. I would approach this by putting `#mrst.sig` into meta. Then when we get [here](https://github.com/apache/couchdb-couch-mrview/blob/master/src/couch_mrview_http.erl#L331), we can use `#mrst.sig` to calculate etag instead of relying on `#vacc.etag` (which is uuid at this point).